File: MyManager/LogManager.vb
Public Class LogManager
    Inherits Manager

    Public Enum MyLevel
        undefined
        MyInfo
        MyWarning
        MyError
    End Enum


    Public Sub New(ByVal connection As System.Data.Common.DbConnection)
        MyBase.New(connection)
    End Sub

    Public Sub New()
        MyBase.New("Log")
    End Sub

    Public Sub New(ByVal connectionName As String)
        MyBase.New(connectionName)
    End Sub

    Public Sub insert(ByVal tipo As String, ByVal nota As String, ByVal referenceId As String, referenceType As String, myLevel As MyLevel)
        Dim strSQL As String
        Dim strSQLParametri As String

        strSQL = "INSERT INTO LOG ( DATE_ADDED  "
        strSQLParametri = " VALUES ( @DATE_ADDED  "

        Dim command As System.Data.Common.DbCommand
        command = mConnection.CreateCommand()
        command.CommandText = strSQL
        command.Connection = mConnection

        Me.mAddParameter(command, "@DATE_ADDED", DateTime.Now)

        If Not String.IsNullOrEmpty(nota) Then
            strSQL &= ",nota "
            strSQLParametri &= ", @nota "
            Me.mAddParameter(command, "@nota", nota)
        End If

        If Not String.IsNullOrEmpty(tipo) Then
            strSQL &= ",tipo "
            strSQLParametri &= ", @tipo "
            Me.mAddParameter(command, "@tipo", tipo)
        End If

        If Not String.IsNullOrEmpty(referenceId) Then
            strSQL &= ",reference_id "
            strSQLParametri &= ", @referenceId "
            Me.mAddParameter(command, "@referenceId", referenceId)
        End If

        If Not String.IsNullOrEmpty(referenceType) Then
            strSQL &= ",reference_type "
            strSQLParametri &= ", @referenceType "
            Me.mAddParameter(command, "@referenceType", referenceType)
        End If

        If myLevel <> LogManager.MyLevel.undefined Then
            strSQL &= ",my_level "
            strSQLParametri &= ", @MY_LEVEL "
            Me.mAddParameter(command, "@MY_LEVEL", myLevel.ToString)
        End If

        command.CommandText = strSQL & " ) " & strSQLParametri & " )"
        command.CommandType = CommandType.Text

        command.ExecuteNonQuery()
    End Sub



    Public Function getLogById(logId As Long) As Data.DataTable
        mStrSQL = "select * from LOG  WHERE LOG_ID = " & logId
        Return mFillDataTable(mStrSQL)
    End Function
End Class

File: MyMessageQueue/MyMessageQueueManager/LogManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace My.MessageQueue
{
    public class LogManager : MyManagerCSharp.ManagerDB
    {
        public enum Level
        {
            Undefined = 0,
            Debug = 1,
            Info = 2,
            Warning = 3,
            Error = 4,
            Exception = 5
        }

        public LogManager(string connectionName)
            : base(connectionName)
        {

        }

        public LogManager(System.Data.Common.DbConnection connection)
            : base(connection)
        {

        }

        public void info(long messageId, string nota)
        {
            _insert(messageId, nota, Level.Info);
        }

        public void warning(long messageId, string nota)
        {
            _insert(messageId, nota, Level.Warning);
        }

        public void error(long messageId, string nota)
        {
            _insert(messageId, nota, Level.Error);
        }

        public void exception(long messageId, Exception ex)
        {
            string temp;
            temp = ex.Message;
            if (ex.InnerException != null)
            {
                temp += Environment.NewLine;
                temp += ex.InnerException.Message;
            }
            _insert(messageId, temp, Level.Exception);
        }

        private void _insert(long messageId, string nota, Level level)
        {
            string strSQLParametri;

            mStrSQL = "INSERT INTO mmq.log ( DATE_ADDED  ";
            strSQLParametri = " VALUES ( GetDate()  ";

            System.Data.Common.DbCommand command;
            command = mConnection.CreateCommand();
            command.Connection = mConnection;


            if (!String.IsNullOrEmpty(nota))
            {
                mStrSQL += ",my_note ";
                strSQLParametri += ", @nota ";
                mAddParameter(command, "@nota", nota);
            }

            mStrSQL += ",message_id ";
            strSQLParametri += ", @messageID ";
            mAddParameter(command, "@messageID", messageId);


            if (level != LogManager.Level.Undefined)
            {
                mStrSQL += ",my_level ";
                strSQLParametri += ", @MY_LEVEL ";
                mAddParameter(command, "@MY_LEVEL", (int)level);
            }

            command.CommandText = mStrSQL + " ) " + strSQLParametri + " )";
            command.CommandType = System.Data.CommandType.Text;

            mExecuteNoQuery(command);
        }




        public List<Models.Log> getLogs(long messageId)
        {
            List<Models.Log> risultato;
            risultato = new List<Models.Log>();

            mStrSQL = "SELECT * FROM  mmq.log WHERE message_id = " + messageId + " order by id";

            mDt = mFillDataTable(mStrSQL);

            foreach (System.Data.DataRow row in mDt.Rows)
            {
                risultato.Add(new Models.Log(row));
            }

            return risultato;
        }

    }
}
